core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+
+
+public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable {
+  private static final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
+  private final Configuration conf;
+  private String warehouseUri;
+
+  public HadoopCatalog(Configuration conf, String warehouseUri) {
+    this.conf = conf;
+
+    if (warehouseUri != null) {
+      this.warehouseUri = warehouseUri;
+    } else {
+      String fsRoot = conf.get("fs.defaultFS");
+      Path warehousePath = new Path(fsRoot, ICEBERG_HADOOP_WAREHOUSE_BASE);
+      try {
+        FileSystem fs = Util.getFs(warehousePath, conf);
+        if (!fs.isDirectory(warehousePath)) {
+          if (!fs.mkdirs(warehousePath)) {
+            throw new IOException("failed to create warehouse for hadoop catalog");
+          }
+        }
+        this.warehouseUri = fsRoot + "/" + ICEBERG_HADOOP_WAREHOUSE_BASE;
+      } catch (IOException e) {
+        throw new RuntimeIOException("failed to create directory for warehouse", e);
+      }
+    }
+  }
+
+  public HadoopCatalog(Configuration conf) {
+    this(conf, null);
+  }
+
+  @Override
+  public org.apache.iceberg.Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, Map<String, String> properties) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+            "Missing database in table identifier: %s", identifier);
+    Path tablePath = new Path(defaultWarehouseLocation(identifier));
+    try {
+      FileSystem fs = Util.getFs(tablePath, conf);
+      if (!fs.isDirectory(tablePath)) {
+        fs.mkdirs(tablePath);
+      } else {
+        throw new AlreadyExistsException("the table already exists: " + identifier);
+      }
+    } catch (IOException e) {
+      throw new RuntimeIOException("failed to create directory", e);
+    }
+    return super.createTable(identifier, schema, spec, null, properties);
+  }
+
+  public org.apache.iceberg.Table createTable(
+          TableIdentifier identifier, Schema schema, PartitionSpec spec) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+            "Missing database in table identifier: %s", identifier);
+    return createTable(identifier, schema, spec, null, null);
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+                "Missing database in table identifier: %s", identifier);
```

Comment:
Why restrict namespaces to 1 level?

Reply:
I think that would be easy to manage the directory. When the level of the namespace is bigger than one, we need to consider whether to create child directories or one directory.

Reply:
I got a misunderstanding here. I have updated the PR to support namespace.